src/Bicep.Core/TypeSystem/Providers/ThirdParty/ThirdPartyResourceTypeProvider.cs

```diff
@@ -0,0 +1,208 @@
+                {
+                    var updatedProperties = new List<TypeProperty>();
+
+                    foreach (var metadataProperty in metadataType.Properties.Values)
+                    {
+                        if (metadataProperty.Name == NamePropertyName || metadataProperty.Name == NamespaceProperty)
+                        {
+                            updatedProperties.Add(metadataProperty);
+                        }
+                        else
+                        {
+                            updatedProperties.Add(new TypeProperty(metadataProperty.Name, metadataProperty.TypeReference, ConvertToReadOnly(metadataProperty.Flags), metadataProperty.Description));
+                        }
+                    }
+
+                    var updatedMetadataType = new ObjectType(
+                        metadataType.Name,
+                        metadataType.ValidationFlags,
+                        updatedProperties,
+                        metadataType.AdditionalPropertiesType,
+                        ConvertToReadOnly(metadataType.AdditionalPropertiesFlags),
+                        functions: null);
+
+                    yield return new TypeProperty(property.Name, updatedMetadataType, property.Flags, property.Description);
+                }
+                else
+                {
+                    yield return new TypeProperty(property.Name, property.TypeReference, ConvertToReadOnly(property.Flags), property.Description);
+                }
+            }
+        }
+
+        private static TypePropertyFlags ConvertToReadOnly(TypePropertyFlags typePropertyFlags)
+            => (typePropertyFlags | TypePropertyFlags.ReadOnly) & ~TypePropertyFlags.Required;
+
+        public ResourceType? TryGetDefinedType(NamespaceType declaringNamespace, ResourceTypeReference typeReference, ResourceTypeGenerationFlags flags)
+        {
+            if (!HasDefinedType(typeReference))
+            {
+                return null;
+            }
+
+            // It's important to cache this result because generating the resource type is an expensive operation
+            var resourceType = definedTypeCache.GetOrAdd(flags, typeReference, () =>
+            {
+                var resourceType = resourceTypeLoader.LoadType(typeReference);
+
+                return SetBicepResourceProperties(resourceType, flags);
+            });
+
+            return new(
+                declaringNamespace,
+                resourceType.TypeReference,
+                resourceType.ValidParentScopes,
+                resourceType.ReadOnlyScopes,
+                resourceType.Flags,
+                resourceType.Body,
+                UniqueIdentifierProperties);
+        }
+
+        //Harsh TODO - placeholder, need to implement this properly
+        public ResourceType? TryGenerateFallbackType(NamespaceType declaringNamespace, ResourceTypeReference typeReference, ResourceTypeGenerationFlags flags)
```

Comment:
Hard-code to return null for now - we're not ready to add fallback types, as this requires a change to the types.json package first.